File: sdk/swift/Sources/PrivacyRPCSDK/PrivacyRPCBranding.swift
import Foundation
#if canImport(UIKit)
import UIKit
#elseif canImport(AppKit)
import AppKit
#endif

/// PrivacyRPC Branding Constants
///
/// Official brand colors, version info, and theming constants
/// matching the PrivacyRPC Chrome Extension and Demo UI.
public enum PrivacyRPCBranding {

    // MARK: - App Identity

    public static let name = "PrivacyRPC"
    public static let version = "2.0.0"
    public static let description = "Privacy-First Solana RPC Protection"
    public static let tagline = "Privacy-first Solana RPC protection"

    // MARK: - Brand Colors (Hex Strings)

    public enum ColorHex {
        /// Primary accent – cyan/teal
        public static let primary = "#5AF5F5"
        public static let primaryHover = "#7FF7F7"
        public static let primaryBg = "#0A1A1A"
        public static let primaryDark = "#0D3D3D"
        public static let primaryBorder = "#1E4040"

        /// Warning – amber
        public static let warning = "#FFB800"
        public static let warningBg = "#1A1508"
        public static let warningBorder = "#3D3A0A"

        /// Danger – red
        public static let danger = "#FF4757"
        public static let dangerBg = "#1A0A0A"
        public static let dangerBorder = "#3D1A1A"

        /// Backgrounds
        public static let background = "#000000"
        public static let surface = "#050505"
        public static let border = "#1E2328"
        public static let surfaceSecondary = "#2A3038"

        /// Text
        public static let textPrimary = "#FFFFFF"
        public static let textMuted = "#7D7D7D"
        public static let textSecondary = "#AAAAAA"
    }

    // MARK: - Native Colors

    #if canImport(UIKit)
    public enum Colors {
        public static let primary        = UIColor(red: 0.353, green: 0.961, blue: 0.961, alpha: 1)
        public static let primaryHover   = UIColor(red: 0.498, green: 0.969, blue: 0.969, alpha: 1)
        public static let primaryBg      = UIColor(red: 0.039, green: 0.102, blue: 0.102, alpha: 1)
        public static let primaryDark    = UIColor(red: 0.051, green: 0.239, blue: 0.239, alpha: 1)
        public static let primaryBorder  = UIColor(red: 0.118, green: 0.251, blue: 0.251, alpha: 1)

        public static let warning        = UIColor(red: 1.0, green: 0.722, blue: 0.0, alpha: 1)
        public static let warningBg      = UIColor(red: 0.102, green: 0.082, blue: 0.031, alpha: 1)
        public static let warningBorder  = UIColor(red: 0.239, green: 0.227, blue: 0.039, alpha: 1)

        public static let danger         = UIColor(red: 1.0, green: 0.278, blue: 0.341, alpha: 1)
        public static let dangerBg       = UIColor(red: 0.102, green: 0.039, blue: 0.039, alpha: 1)
        public static let dangerBorder   = UIColor(red: 0.239, green: 0.102, blue: 0.102, alpha: 1)

        public static let background     = UIColor(red: 0, green: 0, blue: 0, alpha: 1)
        public static let surface        = UIColor(red: 0.02, green: 0.02, blue: 0.02, alpha: 1)
        public static let border         = UIColor(red: 0.118, green: 0.137, blue: 0.157, alpha: 1)

        public static let textPrimary    = UIColor.white
        public static let textMuted      = UIColor(red: 0.49, green: 0.49, blue: 0.49, alpha: 1)
    }
    #elseif canImport(AppKit)
    public enum Colors {
        public static let primary        = NSColor(red: 0.353, green: 0.961, blue: 0.961, alpha: 1)
        public static let primaryHover   = NSColor(red: 0.498, green: 0.969, blue: 0.969, alpha: 1)
        public static let primaryBg      = NSColor(red: 0.039, green: 0.102, blue: 0.102, alpha: 1)
        public static let primaryDark    = NSColor(red: 0.051, green: 0.239, blue: 0.239, alpha: 1)
        public static let primaryBorder  = NSColor(red: 0.118, green: 0.251, blue: 0.251, alpha: 1)

        public static let warning        = NSColor(red: 1.0, green: 0.722, blue: 0.0, alpha: 1)
        public static let warningBg      = NSColor(red: 0.102, green: 0.082, blue: 0.031, alpha: 1)
        public static let warningBorder  = NSColor(red: 0.239, green: 0.227, blue: 0.039, alpha: 1)

        public static let danger         = NSColor(red: 1.0, green: 0.278, blue: 0.341, alpha: 1)
        public static let dangerBg       = NSColor(red: 0.102, green: 0.039, blue: 0.039, alpha: 1)
        public static let dangerBorder   = NSColor(red: 0.239, green: 0.102, blue: 0.102, alpha: 1)

        public static let background     = NSColor(red: 0, green: 0, blue: 0, alpha: 1)
        public static let surface        = NSColor(red: 0.02, green: 0.02, blue: 0.02, alpha: 1)
        public static let border         = NSColor(red: 0.118, green: 0.137, blue: 0.157, alpha: 1)

        public static let textPrimary    = NSColor.white
        public static let textMuted      = NSColor(red: 0.49, green: 0.49, blue: 0.49, alpha: 1)
    }
    #endif

    // MARK: - Status Labels

    public enum Status {
        public static let active = "Active"
        public static let inactive = "Inactive"
        public static let protected_ = "Protected"
        public static let proxyOffline = "Proxy Offline"
        public static let torActive = "Tor Active"
        public static let publicRpc = "Public RPC"
        public static let connecting = "Connecting..."
    }
}

File: sdk/kotlin/src/main/kotlin/com/privacyrpc/sdk/PrivacyRPCBranding.kt
package com.privacyrpc.sdk

/**
 * PrivacyRPC Branding Constants
 *
 * Official brand colors, version info, and theming constants
 * matching the PrivacyRPC Chrome Extension and Demo UI.
 */
object PrivacyRPCBranding {

    // ── App Identity ──────────────────────────────────────────
    const val NAME = "PrivacyRPC"
    const val VERSION = "2.0.0"
    const val DESCRIPTION = "Privacy-First Solana RPC Protection"
    const val TAGLINE = "Privacy-first Solana RPC protection"

    // ── Brand Colors (Hex) ────────────────────────────────────
    object Colors {
        /** Primary accent – cyan/teal */
        const val PRIMARY = "#5AF5F5"
        /** Primary hover */
        const val PRIMARY_HOVER = "#7FF7F7"
        /** Primary background tint */
        const val PRIMARY_BG = "#0A1A1A"
        /** Primary dark tint */
        const val PRIMARY_DARK = "#0D3D3D"
        /** Primary border */
        const val PRIMARY_BORDER = "#1E4040"

        /** Warning – amber */
        const val WARNING = "#FFB800"
        /** Warning background tint */
        const val WARNING_BG = "#1A1508"
        /** Warning border */
        const val WARNING_BORDER = "#3D3A0A"

        /** Danger – red */
        const val DANGER = "#FF4757"
        /** Danger background tint */
        const val DANGER_BG = "#1A0A0A"
        /** Danger border */
        const val DANGER_BORDER = "#3D1A1A"

        /** Background – pure black */
        const val BACKGROUND = "#000000"
        /** Card / surface */
        const val SURFACE = "#050505"
        /** Border / separator */
        const val BORDER = "#1E2328"
        /** Secondary surface */
        const val SURFACE_SECONDARY = "#2A3038"

        /** Text – white */
        const val TEXT_PRIMARY = "#FFFFFF"
        /** Muted text */
        const val TEXT_MUTED = "#7D7D7D"
        /** Secondary text */
        const val TEXT_SECONDARY = "#AAAAAA"
    }

    // ── Brand Colors (ARGB Int for Android) ──────────────────
    object ColorsInt {
        const val PRIMARY: Long          = 0xFF5AF5F5
        const val PRIMARY_HOVER: Long    = 0xFF7FF7F7
        const val PRIMARY_BG: Long       = 0xFF0A1A1A
        const val PRIMARY_DARK: Long     = 0xFF0D3D3D
        const val PRIMARY_BORDER: Long   = 0xFF1E4040

        const val WARNING: Long          = 0xFFFFB800
        const val WARNING_BG: Long       = 0xFF1A1508
        const val WARNING_BORDER: Long   = 0xFF3D3A0A

        const val DANGER: Long           = 0xFFFF4757
        const val DANGER_BG: Long        = 0xFF1A0A0A
        const val DANGER_BORDER: Long    = 0xFF3D1A1A

        const val BACKGROUND: Long       = 0xFF000000
        const val SURFACE: Long          = 0xFF050505
        const val BORDER: Long           = 0xFF1E2328
        const val SURFACE_SECONDARY: Long = 0xFF2A3038

        const val TEXT_PRIMARY: Long     = 0xFFFFFFFF
        const val TEXT_MUTED: Long       = 0xFF7D7D7D
    }

    // ── Typography ───────────────────────────────────────────
    object Typography {
        const val FONT_FAMILY = "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif"
        const val MONO_FONT = "'SF Mono', Monaco, monospace"
    }

    // ── Status Labels ────────────────────────────────────────
    object Status {
        const val ACTIVE = "Active"
        const val INACTIVE = "Inactive"
        const val PROTECTED = "Protected"
        const val PROXY_OFFLINE = "Proxy Offline"
        const val TOR_ACTIVE = "Tor Active"
        const val PUBLIC_RPC = "Public RPC"
        const val CONNECTING = "Connecting..."
    }
}
